src/Features/LanguageServer/Protocol/Handler/RequestContext.cs

```diff
@@ -66,6 +68,15 @@ namespace Microsoft.CodeAnalysis.LanguageServer.Handler
 
                 if (!documents.IsEmpty)
                 {
+                    if (workspace.IsMiscellaneousFilesWorkspace)
+                    {
+                        Logger.Log(FunctionId.FindDocument_InMiscellaneousFilesWorkspace);
+                    }
+                    else
+                    {
+                        Logger.Log(FunctionId.FindDocument_InWorkspace);
```

Comment:
I'm wondering if it would be easier to write a query based off a single event with a property in the bag that gives the workspace type.  That way when we register for example the anycode workspace, we can get a breakdown of Misc vs. VSworkspace vs. anycode workspace.

Anycode is probably slightly common in codespaces as unless there is a single sln, folder view will be the default view.  this can affect some feature functionality as well, so it would be good to know.

then we also don't need to add the new property to the workspace

Reply:
sounds good, my only concern is the protocol layer has no way to ask a workspace it "type" today unless there is something I missed?

Reply:
nevermind workspace.kind should be sufficient